MinitexPDFProtocols/MinitexPDFProtocols.swift

```diff
@@ -9,67 +9,92 @@
-  @objc func saveAnnotations(annotations: [MinitexPDFAnnotation])
-  @objc func willMoveToMinitexParentController(parent: UIViewController?)
+
+  /// Called when a page of the PDF document becomes the new presented page
+  /// visible on screen.
+  ///
+  /// - Parameter page: The page being presented on screen.
+  func userDidNavigate(toPage page: MinitexPDFPage)
+  /// Called when a user selects to create a bookmark for a specific page.
+  ///
+  /// - Parameter bookmark: The specific page the user has selected to save as a
+  ///   bookmark.
+  func userDidCreate(bookmark: MinitexPDFPage)
+  /// Called when a user selects to delete a bookmark for a specific page.
+  ///
+  /// - Parameter bookmark: The specific page the user has selected to delete.
+  func userDidDelete(bookmark: MinitexPDFPage)
+  /// Called when a user selects to create a generic annotation for a specific page.
+  ///
+  /// - Parameter bookmark: The specific generic annotation the user has selected to save as a
+  ///   bookmark.
+  func userDidCreate(annotation: MinitexPDFAnnotation)
+  /// Called when a user selects to delete a generic annotation for a specific page.
+  ///
+  /// - Parameter bookmark: The specific generic annotation the user has selected to delete.
+  func userDidDelete(annotation: MinitexPDFAnnotation)
 }
 
+//MARK: -
+//MARK: Factory Methods
 
-@objc public class MinitexPDFViewControllerFactory: NSObject {
-  @objc public static func createPDFViewController(dictionary: [String: Any]) -> MinitexPDFViewController? {
+@objcMembers public class MinitexPDFViewControllerFactory: NSObject {
 
-    guard let pdfViewControllerClass = NSClassFromString("PDF.PDFViewController")
-                                        as? MinitexPDFViewController.Type else {
-      return nil
-    }
+  /// The API host must create a View Controller for the PDF from a static factory method.
+  ///
+  /// - Parameters:
+  ///   - fileUrl: file directory of the PDF document
+  ///   - page: optional page to open to
+  ///   - bookmarks: any saved bookmarks by the user
+  ///   - annotations: any additional generic annotations created by the user
+  /// - Returns: An instance of MinitexPDFViewController, or nil if something failed.
+  public static func create(fileUrl: URL,
+                            openToPage page: MinitexPDFPage,
+                            bookmarks: [MinitexPDFPage],
+                            annotations: [MinitexPDFAnnotation]) -> MinitexPDFViewController? {
 
-    let pdfViewController = pdfViewControllerClass.init(dictionary: dictionary)
-    return pdfViewController
+    // PDF renderer using a PSPDFKit license.
+    if let pdfViewControllerClass = NSClassFromString("PDF.PDFViewController") as? MinitexPDFViewController.Type {
+      return pdfViewControllerClass.init(file: fileUrl, openToPage: page, bookmarks: bookmarks, annotations: annotations)
+    } else {
+      // PDF renderer using Apple's PDFKit.
```

Comment:
Just like in Dean's audiobook frameworks, there will be a generic implementation of PDF support provided if no other renderer is present.